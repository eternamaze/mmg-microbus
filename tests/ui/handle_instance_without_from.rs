use mmg_microbus::prelude::*;

#[mmg_microbus::component]
struct C { id: mmg_microbus::bus::ComponentId }

#[mmg_microbus::handles]
impl C {
    #[mmg_microbus::handle(Tick, instance="x")] // ERROR: instance without from
    async fn on_tick(&mut self, _env: std::sync::Arc<Envelope<Tick>>) -> Result<()> { Ok(()) }
}

#[derive(Clone)]
struct Tick;

fn main() {}
