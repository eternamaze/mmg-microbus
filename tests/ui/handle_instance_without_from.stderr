error: #[handle]: `instance = ...` requires also specifying `from = ServiceType`
 --> tests/ui/handle_instance_without_from.rs:6:1
  |
6 | #[mmg_microbus::handles]
  | ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `mmg_microbus::handles` (in Nightly builds, run with -Z macro-backtrace for more info)
